### PR TITLE
Scripts: Use cssnano to minimize CSS files with build

### PR DESCRIPTION
**Pull request by gziolo**
## Description

Follow-up for #33676 and the comment from @ocean90 https://github.com/WordPress/gutenberg/pull/33676#issuecomment-888435234:

> This should probably be released as a breaking change because it can impact users which have their own PostCSS config which already includes cssnano (which optimize-css-assets-webpack-plugin uses by default) with a custom config for [allowed optimisations](https://cssnano.co/docs/optimisations/).
> 
> Since sass-loader seems to minify the CSS already, should we maybe extend the default PostCSS config to use cssnano instead?
> 
> Related:
> * [github.com/WordPress/gutenberg/blob/44212878dcbf6f4616266eb0fb4756293a471615/tools/webpack/shared.js#L82-L93](https://github.com/WordPress/gutenberg/blob/44212878dcbf6f4616266eb0fb4756293a471615/tools/webpack/shared.js#L82-L93)
> * [github.com/WordPress/gutenberg/blob/44212878dcbf6f4616266eb0fb4756293a471615/packages/scripts/config/webpack.config.js#L47-L55](https://github.com/WordPress/gutenberg/blob/44212878dcbf6f4616266eb0fb4756293a471615/packages/scripts/config/webpack.config.js#L47-L55)
> * [github.com/WordPress/gutenberg/blob/44212878dcbf6f4616266eb0fb4756293a471615/packages/postcss-plugins-preset/lib/index.js](https://github.com/WordPress/gutenberg/blob/44212878dcbf6f4616266eb0fb4756293a471615/packages/postcss-plugins-preset/lib/index.js)

## How has this been tested?

I created a custom block with:

```bash
npx wp-create-block example --no-wp-scripts
cd example
../node_modules/.bin/wp-scripts build
```

I updated used CSS files to end with `.css` to ensure that the output is properly minimized.

I also ensured that the output of CSS files isn't minified in the dev mode with:
```bash
../node_modules/.bin/wp-scripts start
```

I added `cssnano.config.js` file in the root folder of the project with the following content:

```js
module.exports = {
	preset: [ require( 'cssnano-preset-default' ), { discardComments: false } ],
};
```

I also had to install `cssnano-preset-default' with:

```bash
npm i cssnano-preset-default --save-dev
```

When I run:

```bash
../node_modules/.bin/wp-scripts build
```

I can see that comments in the CSS files are preserved. It means that the custom config is properly loaded when present.

## Types of changes

Enhancement.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/scripts/config/webpack.config.js [hunk 1]
```diff
@@ -49,7 +48,10 @@ const cssLoaders = [
 			...( ! hasPostCSSConfig() && {
 				postcssOptions: {
 					ident: 'postcss',
-					plugins: postcssPlugins,
+					sourceMap: ! isProduction,
+					plugins: isProduction
+						? [ ...postcssPlugins, require( 'cssnano' )() ]
```
Comment (gziolo) [inline packages/scripts/config/webpack.config.js]: @ocean90, should we provide some default configuration for cssnano?
We could add a similar check as for PostCSS config, there are some ways to include the config for cssnano according to https://cssnano.co/docs/config-file/.

File: packages/scripts/config/webpack.config.js [hunk 2]
```diff
@@ -49,7 +48,10 @@ const cssLoaders = [
 			...( ! hasPostCSSConfig() && {
 				postcssOptions: {
 					ident: 'postcss',
-					plugins: postcssPlugins,
+					sourceMap: ! isProduction,
```
Comment (gziolo) [inline packages/scripts/config/webpack.config.js:52]: It aligns with the setup for CSS loader.

File: packages/scripts/config/webpack.config.js (hunk 1 above)
Comment (ocean90) [reply packages/scripts/config/webpack.config.js]: I think we can use the same preset as used in https://github.com/WordPress/gutenberg/blob/44212878dcbf6f4616266eb0fb4756293a471615/tools/webpack/shared.js#L84-L91
([Related docs](https://github.com/cssnano/cssnano/tree/master/packages/postcss-discard-comments#removeall) for `removeAll`)

If we go with this adding a check like `hasPostCSSConfig()` makes sense.

File: packages/scripts/utils/config.js [hunk 3]
```diff
@@ -24,6 +24,16 @@ const hasBabelConfig = () =>
 	hasProjectFile( '.babelrc' ) ||
 	hasPackageProp( 'babel' );
 
+// See https://cssnano.co/docs/config-file.
+const hasCssnanoConfig = () =>
```
Comment (gziolo) [inline packages/scripts/utils/config.js:28]: I discovered in the source code that the list of supported file names is longer that in the docs.

**Review by ocean90 — APPROVED** [review]
👍 LGTM

**Comment by shivapoudel** [comment]
@gziolo Wasn't this CSSNano feature included in 17.1.0. I couldn't achieve it!

**Comment by gziolo** [comment]
No, it's still in the development `trunk` branch only. We try to publish to npm every two weeks based on the codebase that gets released in the Gutenberg plugin.

**Comment by getdave** [comment]
@gziolo I noticed this PR (and one other) appearing under "Enhancements" in the auto-changelog. I wonder is there a better label we could assign to changes like this that are "Tooling" but not necessarily _Build_ Tooling?

**Comment by gziolo** [comment]
@getdave, I'm open to improvements in this area. Sometimes it's very hard to match existing labels with work that doesn't impact production code. This PR is tricky as well, it was an existing behavior when using `.scss` files, so it's even hard to tell if this should be a new feature (for `.css` files) or an enhancement or maybe a bug fix because we missed that.

I'm sure about one change, we should at least rename `Build Tooling` to `Tooling` so it could cover more use cases. A more granular label could be `Build Process`.

**Pull request merged**